src/scrapers/base-scraper-with-browser.ts

```diff
@@ -169,6 +131,57 @@ class BaseScraperWithBrowser<TCredentials extends ScraperCredentials> extends Ba
     });
   }
 
+  private async initializePage() {
+    debug('initialize browser page');
+    if ('browserContext' in this.options) {
+      debug('Using the browser context provided in options');
+      return this.options.browserContext.newPage();
+    }
+
+    if ('browser' in this.options) {
+      debug('Using the browser instance provided in options');
+      const { browser } = this.options;
+
```

Comment:
I need to review it, but I don't have a time for it

Reply:
@baruchiro, I aggrege that reviewing large changes in GH mobile is hard.
Attached here is the `o1` answer for the changes, LMK if it makes the review easyer.
__________

_AI Generated:_

<details>
<summary>
Longe explanation:
</summary>

### **Changes Introduced:**

- **Added Support for External `BrowserContext`:**
  - **New Logic Added:**
    - In the `initializePage()` method, we now first check if a `browserContext` is provided in `this.options`.
    - **If a `browserContext` is provided:**
      - We create a new page using `browserContext.newPage()`.
      - This allows users to supply their own `BrowserContext`, offering more flexibility when the browser context is managed externally.

---

### **Logic Unchanged:**

- **Existing Browser Initialization Logic Remains the Same:**
  - **If no `browserContext` is provided:**
    - We proceed to check if a `browser` is provided in `this.options` (same as before).
      - **If a `browser` is provided:**
        - We log a debug message: `'Using the browser instance provided in options'`.
        - We create a new page using `browser.newPage()`.
        - For backward compatibility, if `skipCloseBrowser` is not set to `true`, we add a cleanup function to close the browser after scraping.
      - **If no `browser` is provided:**
        - We launch a new browser instance as before, using options like `executablePath`, `args`, `showBrowser`, and `timeout` from `this.options`.
        - We add a cleanup function to close the browser after scraping.
        - If `prepareBrowser` is provided, we call it with the new browser instance.

- **Resource Cleanup Logic Remains the Same:**
  - We maintain a `cleanups` array to manage cleanup functions.
  - During termination (`terminate()` method), we execute all cleanup functions in reverse order to ensure proper resource disposal.
  - This includes closing pages and browsers as needed.
  
- **Page Creation and Return:**
  - After handling the browser or browser context, we create a new page and return it.
  - This part of the logic is unchanged.

</details>

### **Summary:**

- **What's New:**
  - Added support for an external `BrowserContext` in `initializePage()`, allowing more flexibility in browser management.

- **What's Unchanged:**
  - The existing logic for:
    - Handling an external `Browser` instance.
    - Launching a new browser if none is provided.
    - Preparing the browser with `prepareBrowser` if provided.
    - Managing resource cleanup with the `cleanups` array.
    - Creating and returning a new page for the scraper to use.
  - All existing behavior remains intact, ensuring backward compatibility.